build.assets/Dockerfile

```diff
@@ -40,11 +32,16 @@ RUN (mkdir -p /opt/protoc && \
      git clone https://github.com/gogo/protobuf --branch ${GOGO_PROTO_TAG} /src/github.com/gogo/protobuf && cd /src/github.com/gogo/protobuf && make install && \
      git clone https://github.com/grpc-ecosystem/grpc-gateway --branch ${GRPC_GATEWAY_TAG} /src/github.com/grpc-ecosystem/grpc-gateway && cd /src/github.com/grpc-ecosystem/grpc-gateway && pwd && go install ./protoc-gen-grpc-gateway)
 
-ENV PROTO_INCLUDE "/usr/local/include":"/src":"${GRPC_GATEWAY_ROOT}/third_party/googleapis":"${GOGOPROTO_ROOT}/gogoproto"
+ARG USER_ID=1000
+RUN adduser jenkins --uid=$USER_ID --disabled-password --system
+RUN (mkdir -p /src/github.com/gravitational/gravity && \
+     chown -R jenkins /gopath && \
+     chown -R jenkins /src && \ 
+     mkdir -p /.cache && \
+     chmod 777 /.cache && \
+     chmod 777 /tmp)
```

Comment:
Won't putting these RUN directives after the clones result in a bunch of superfulous layer churn and ever so slightly slower builds?  I'd expect because these are relatively static, they should be before the dynamic `git clone` layers, where they were originally.

Reply:
Hmm, I didn't think about that. I moved all that code because of `chown -R jenkins /gopath`, otherwise after building grpc few lines above everything inside gopath is owned by root. I will implement the fix for that situation differently. Thanks for pointing on that.